### DISCUSSION TITLE
Download quota per acct addon

### DISCUSSION THREAD

**Discussion by Stoufs** [rejetto/hfs#428] [General]
Is it possible to add a feature that auto disables the account after going through their quota automatically?

**Comment by rejetto** [comment]
I guess you forgot, but you already asked :) https://github.com/rejetto/hfs/discussions/422#discussioncomment-7974641

**Comment by Stoufs** [comment]
What about expire 30 days from first login. Regardless of quota?

**Comment by Stoufs** [comment]
Hey! What about auto disabling account after 30 days from first time its
logged into regardless of quota

**Comment by Stoufs** [comment]
Or 30 days from account creation

**Comment by rejetto** [comment]
and what to do with the account when the time comes? delete it or just prevent it from working?
and would it be preferable to start with account creation or first login?

**Comment by Stoufs** [comment]
Time of login sorry and delete account after 30 days from time of login regardless of quota left. Maybe a tick option in account creation to auto delete or disable after x amount of days.  Either one works but give the ability to have those accounts not affected if i choose to keep em permanently open

**Comment by rejetto** [comment]
I was thinking of having an optional expiration on any account, as part of HFS.
The rest may be better as plugin, I will think about it.
Do you plan to add accounts manually from the Admin-panel, right?

**Comment by Stoufs** [comment]
Correct

**Comment by Stoufs** [comment]
The download counter seems to be counting the entirety of the file size even if its only downloading a portion. Say the file is 60gb but customer downloads only 5 gb then that acts against them and still counts the entire 60 instead of the actual download. Is there anyway to fix this?

Also is it possible to see how much data each user has downloaded?

**Comment by Stoufs** [comment]
The download counter seems to be counting the entirety of the file size
even if its only downloading a portion. Say the file is 60gb but customer
downloads only 5 gb then that acts against them and still counts the entire
60 instead of the actual download. Is there anyway to fix this? Also is it
possible to see how much data each user has downloaded?

**Comment by rejetto** [comment]
i see... i guess i still have to consume the quota ahead, to reserve for the download that's ongoing, and only then restore the unused quota if it is terminated early.
i will have to study the thing to see how exactly.

**Comment by rejetto** [comment]
to look at each user's quota you can see the file "per-account.json" that is created in the plugin's storage folder

**Comment by rejetto** [comment]
i may have solved the problem, and you can test the version 0.22 of the plugin
